publication-validator/build.gradle

```diff
@@ -2,34 +2,33 @@
  * Copyright 2016-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
 
-plugins {
-    id 'org.jetbrains.kotlin.jvm' version '1.3.30'
-}
-
-def deployVersion = properties['DeployVersion']
-ext.coroutines_version = deployVersion
-println "Checking coroutines version $coroutines_version"
-
-group 'org.jetbrains.kotlinx'
-version '1.0-SNAPSHOT'
+apply from: rootProject.file("gradle/compile-jvm.gradle")
 
 repositories {
     mavenLocal()
     mavenCentral()
 }
 
 dependencies {
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
+    testCompile "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
     testCompile 'junit:junit:4.12'
     testCompile 'org.apache.commons:commons-compress:1.18'
     testCompile 'com.google.code.gson:gson:2.8.5'
-    testCompile "org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutines_version"
-    testCompile "org.jetbrains.kotlinx:kotlinx-coroutines-android:$coroutines_version"
+    testCompile project(':kotlinx-coroutines-core')
+    testCompile project(':kotlinx-coroutines-android')
 }
 
-compileKotlin {
-    kotlinOptions.jvmTarget = "1.8"
-}
 compileTestKotlin {
     kotlinOptions.jvmTarget = "1.8"
 }
+
+task testPublishing(type: Test) {
```

Comment:
Could you please elaborate on why this task does not depend on the root `build` task?

Reply:
The intention was that running `testPublishing` does the same as what is currently being done in `Dependency validation` configuration on TeamCity:
* `--info -PdryRun=true publishNpm publishToMavenLocal -x dokka` in the root project;
* `build` in the project of the validator.
So, if there was a reason to depend on `build`, it wasn't reflected in the build process before.

If we don't appeal to how it's currently done, this still seems reasonable to me: to test publication, we don't really need to know that the stress tests pass, for example; we only need to see the to-be-published packages themselves, and explicitly only depending on them being prepared for publishing conveys this intention more clearly.

Reply:
I changed the configuration so that `build` depends on this validation.